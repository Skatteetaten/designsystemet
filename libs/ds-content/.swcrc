{
  "jsc": {
    "target": "es2021",
    "parser": {
      "syntax": "typescript",
      "tsx": true,
      "decorators": true,
      "dynamicImport": true
    },
    "transform": {
      "react": {
        "runtime": "automatic",
        "importSource": "react"
      },
      "decoratorMetadata": true,
      "legacyDecorator": true
    },
    "keepClassNames": true,
    "externalHelpers": true,
    "loose": true
  },
  "module": {
    "type": "es6"
  },
  "sourceMaps": true,
  "exclude": ["jest.config.ts",".*\\.spec.tsx?$",".*\\.test.tsx?$","./src/jest-setup.ts$","./**/jest-setup.ts$",".*.js$"]
}
