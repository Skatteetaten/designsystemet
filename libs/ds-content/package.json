{
  "name": "@skatteetaten/ds-content",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.4.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18",
    "@skatteetaten/ds-core-utils": "1.4.0",
    "@skatteetaten/ds-typography": "1.4.0",
    "@skatteetaten/ds-icons": "1.4.0",
    "react-i18next": "^14",
    "i18next": "^23"
  },
  "scripts": {},
  "dependencies": {},
  "devDependencies": {}
}
