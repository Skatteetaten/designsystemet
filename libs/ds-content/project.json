{
  "name": "ds-content",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-content/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-content",
        "tsConfig": "libs/ds-content/tsconfig.lib.json",
        "project": "libs/ds-content/package.json",
        "entryFile": "libs/ds-content/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-content/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-content/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-content/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-content/**/*.css",
          "libs/ds-content/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-content"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-content {args.switches}"
      }
    }
  }
}
