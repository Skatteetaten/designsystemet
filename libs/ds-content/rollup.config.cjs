const path = require('path');

const createRollupConfig = require('../ds-dev-config/build/rollup-config-factory.cjs');

const srcDir = path.resolve(__dirname, `src`);
const pathToTsConfig = 'libs/ds-content/tsconfig.lib.json';
const pathToCSS = path.resolve(__dirname, `**/*.{css,scss}`);
const outputDir = path.resolve('./', `dist/libs/ds-content/`);

module.exports = (config) =>
  createRollupConfig(config, srcDir, pathToTsConfig, pathToCSS, outputDir);
