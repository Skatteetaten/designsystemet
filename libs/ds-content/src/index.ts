export * from './Card/Card';
export * from './Card/Card.types';
export * from './CardAlert/CardAlert.types';
export * from './DescriptionList/DescriptionList';
export * from './DescriptionList/DescriptionList.types';
export * from './Panel/Panel';
export * from './Panel/Panel.types';
