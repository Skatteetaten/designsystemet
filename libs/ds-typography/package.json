{
  "name": "@skatteetaten/ds-typography",
  "version": "1.1.0",
  "groupId": "no.skatteetaten.aurora",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.1.0"
  },
  "scripts": {}
}
