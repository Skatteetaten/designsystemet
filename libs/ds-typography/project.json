{
  "name": "ds-typography",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-typography/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-typography",
        "tsConfig": "libs/ds-typography/tsconfig.lib.json",
        "project": "libs/ds-typography/package.json",
        "entryFile": "libs/ds-typography/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-typography/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-typography/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-typography/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-typography/**/*.css",
          "libs/ds-typography/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-typography"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish  dist/libs/ds-typography {args.switches}"
      }
    }
  }
}
