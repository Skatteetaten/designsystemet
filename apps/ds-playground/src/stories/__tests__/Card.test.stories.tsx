import { StoryFn, Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { Card, cardColorArr } from '@skatteetaten/ds-content';
import { Heading, Paragraph } from '@skatteetaten/ds-typography';

const meta = {
  component: Card,
  title: 'Tester/Card/Card',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: 'text',
    },
    // Aria
    ariaLabelledBy: { table: { disable: true } },
    color: { table: { disable: true } },
  },
} satisfies Meta<typeof Card>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultText =
  'Gruppering av opplysninger i skjema, for eksempel inntekter, personer eller oppsummeringer.';

const defaultArgs = {
  children: defaultText,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'en',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const card = canvas.getAllByRole('generic')[1];
    await expect(card).toHaveClass('dummyClassname');
    await expect(card).toHaveAttribute('id', 'htmlId');
    await expect(card).toHaveAttribute('lang', 'en');
    await expect(card).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (B1, B2)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    children: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];

    const section = canvas.queryByRole('region');
    await expect(section).not.toBeInTheDocument();
    await expect(container).toBeInTheDocument();
  },
} satisfies Story;

const TemplateAllColors: StoryFn<typeof Card> = (args) => (
  <>
    {cardColorArr.map((color, index) => {
      return (
        <Card
          {...args}
          key={`card_${index}`}
          className={'bottomSpacingXL'}
          color={color}
        >
          <Card.Content>{args.children}</Card.Content>
        </Card>
      );
    })}
  </>
);

export const AllColors = {
  render: TemplateAllColors,
  name: 'All Colors (A1, A2)',
  args: {
    ...defaultArgs,
  },
} satisfies Story;

export const WithAriaLabelledBy = {
  name: 'With aria-labelledby (B1)',
  argTypes: {
    ariaLabelledBy: { table: { disable: false } },
  },
  args: {
    ...defaultArgs,
    ariaLabelledBy: 'dummyId',
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const cardNote = canvas.getByRole('region');
    await expect(cardNote).toHaveAttribute('aria-labelledby', 'dummyId');
  },
} satisfies Story;
