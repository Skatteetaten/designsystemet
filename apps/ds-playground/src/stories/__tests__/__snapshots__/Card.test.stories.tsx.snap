// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Card/Card AllColors smoke-test 1`] = `
<div data-test-block="true">
  <div class="_card_1x4sk_5  bottomSpacingXL"
       data-color="ochre"
  >
    <div class="_cardContent_1i00x_1 ">
      <div>
        Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
      </div>
    </div>
  </div>
  <div class="_card_1x4sk_5  bottomSpacingXL"
       data-color="forest"
  >
    <div class="_cardContent_1i00x_1 ">
      <div>
        Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
      </div>
    </div>
  </div>
  <div class="_card_1x4sk_5  bottomSpacingXL"
       data-color="burgundy"
  >
    <div class="_cardContent_1i00x_1 ">
      <div>
        Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
      </div>
    </div>
  </div>
  <div class="_card_1x4sk_5  bottomSpacingXL"
       data-color="denim"
  >
    <div class="_cardContent_1i00x_1 ">
      <div>
        Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
      </div>
    </div>
  </div>
  <div class="_card_1x4sk_5  bottomSpacingXL"
       data-color="graphite"
  >
    <div class="_cardContent_1i00x_1 ">
      <div>
        Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
      </div>
    </div>
  </div>
  <div class="_card_1x4sk_5  bottomSpacingXL"
       data-color="white"
  >
    <div class="_cardContent_1i00x_1 ">
      <div>
        Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Card/Card Defaults play-test 1`] = `
<div data-test-block="true">
  <div class="_card_1x4sk_5"
       data-color="white"
  >
    Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
  </div>
</div>
`;

exports[`Tester/Card/Card WithAriaLabelledBy play-test 1`] = `
<div data-test-block="true">
  <section class="_card_1x4sk_5"
           data-color="white"
           aria-labelledby="dummyId"
  >
    Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
  </section>
</div>
`;

exports[`Tester/Card/Card WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       class="_card_1x4sk_5  dummyClassname"
       lang="en"
       data-testid="123ID"
       data-color="white"
  >
    Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
  </div>
</div>
`;

exports[`Tester/Card/Card WithRef play-test 1`] = `
<div data-test-block="true">
  <div class="_card_1x4sk_5"
       data-color="white"
       id="dummyIdForwardedFromRef"
  >
    Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.
  </div>
</div>
`;
